src/pygama/evt/modules/xtalk.py

```diff
@@ -0,0 +1,314 @@
+"""
+Module for cross talk correction of energies.
+"""
+
+from __future__ import annotations
+
+import awkward as ak
+import numpy as np
+from legendmeta.catalog import Props
+from lgdo import lh5, types
+from lgdo.lh5 import ls
+
+from pygama.hit.build_hit import _remove_uneeded_operations, _reorder_table_operations
+
+from .. import utils
+
+
+def build_tcm_id_array(
```

Comment:
The naming of the TCM stuff is confusing, please rename as:
- `tcm_id` if it's the rawids (`tcm.id`). Can also just use `rawid`
- `tcm_idx` if it's the row in the hit-oriented tier files (`tcm.idx`). Can also use `hit_idx` (this is the name in the evt files)
- `tcm_index` if it's the row of the TCM arrays themselves (this is used in the evt config to filter other fields)

Reply:
I would rename this function to sth like `gather_tcm_indices()` in analogy with `spms.gather_tcm_data()`?

Reply:
changed it to ``tcm_index'' throughout